src/java.base/share/classes/com/sun/crypto/provider/CipherCore.java

```diff
@@ -722,8 +723,8 @@ int update(byte[] input, int inputOffset, int inputLen, byte[] output,
         len = (len > 0 ? (len - (len % unitBytes)) : 0);
 
         // check output buffer capacity
-        if ((output == null) ||
-            ((output.length - outputOffset) < len)) {
+        if (getMode() != GCM_MODE && (output == null ||
```

Comment:
Is the buffer size for GCM checked somewhere else? For decryption, it's probably ok to skip the check as no output is returned until doFinal(...). For encryption, there is still a chance for ShortBufferException? Javadoc states that for ShortBufferException, the call can be re-tried with a larger buffer.

Reply:
I know the short buffers were being checked, but I removed the mode check, the tests still pass fine, so I figured this must be a unnecessary change.